### ISSUE TITLE
Наследование от блока с модификатором

### ISSUE THREAD

**Issue by Oopscurity** [bem/bem-forum-content-ru#855]
Пробовал в списке зависимостей указывать `link_animated` - не вышло. Подобным образом — то же самое.
```javascript
modules.define(
	'scroll-top',
	[ 'i-bem__dom', 'jquery' ],
	function(provide, BEMDOM, $) {
		provide(BEMDOM.decl({ block: this.name, baseBlock: { block: 'link', modName: 'animated', modVal: true } }, {
			// ...
		}));
	}
);
```
Возможно осуществить задуманное?

**Comment by tadatuta** [comment]
Нет, этого «не нужно хотеть», т.к. модификатор — это свойство рантайма: он может появиться или исчезать на блоке в процессе взаимодействия с пользователем.

**Comment by Oopscurity** [comment]
Как же, к примеру, `button` с модификатором `type` значения `submit` — разве это свойство рантайма? У меня примерно та же ситуация: модификатор не требуется убирать в рантайме.

**Comment by tadatuta** [comment]
Это частный случай и имеет отношение скорее к шаблонам, чем к JS.

С точки зрения JS любой модификатор *потенциально* можно удалить в любой момент.

**Comment by Oopscurity** [comment]
Тогда если модификатор "не отзывчивый", а функционально от данной разновидности блока может наследоваться что-то иное, то следует оформлять новый блок: к примеру, `animated-link` — я правильно понял? (а далее требуется лишь наследоваться от простого блока)

**Comment by tadatuta** [comment]
Как вариант — да. Еще можно подумать в сторону композиции.
Если что-то ведет себя как анимированная ссылка **и** умеет подскроллиться вверх, то, возможно, это просто одновременно и ссылка и блок про скролл:
```html
<a class="link link_animated scroll-top"></a>
```

**Comment by zxqfox** [comment]
Вообще говоря, глубокие цепочки наследований в самом ООП сложно поддерживать — даже со внуками уже сложно, кроме того, еще и чистый ООП в javascript ущербный, нет чистых приватных свойств (вроде как и не нужны), как и сложно реализовать protected (их все равно видно, но можно их переименовать).

Но еще можно посмотреть в сторону миксов, @tadatuta они же есть? Я только не знаю как и православно оформлять в виде блоков.

**Comment by ilyar** [comment]
> Но еще можно посмотреть в сторону миксов

@zxqfox речь об этом?

```JavaScript
/* common.blocks/link/_animated/link_animated.js */
modules.define('link_animated', ['i-bem__dom'], function(provide, BEMDOM) {

/**
 * @exports
 * @class link_animated
 */
provide(BEMDOM.declMix('link_animated', // только строка с именем
    /** @lends link_animated.prototype */ {},
    /** @lends link_animated */ {}
));

});

/* common.blocks/scroll-top/scroll-top.js */
modules.define('scroll-top', ['i-bem__dom', 'link_animated'], function(provide, BEMDOM) {

/**
 * @exports
 * @class scroll-top
 */
provide(BEMDOM.decl({ block : this.name, baseMix : ['link_animated']},
    /** @lends scroll-top.prototype */ {},
    /** @lends scroll-top */ {}
));

});
```

**Comment by zxqfox** [comment]
Да, об этом. В такой микс можно вынести общую оторванную функциональность и после примешать обоим блокам.
